### DISCUSSION TITLE
Control tiles size and composite

### DISCUSSION THREAD

**Discussion by gijdoul** [ImageMagick/ImageMagick#2206] [Help]
Hello,

I have seamless patterns with 3600x3600px and  I need to create a canvas with this size 2550x1729px . What I'm trying to do is fill my pattern to the canvas I found a way to do it but the result is the pattern is too big I want to know to reduce the pattern 
size on the new canvas.

I try this:  **composite -tile mypattern.jpg -size 2550x1729  xc:none   mycanvas.jpg**

but don't know how to add tile pattern size parameter before 

Thanks for your help

**Comment by snibgo** [comment]
I don't understand what you want. Your pattern is smaller than the canvas you want. So if you want to repeat the pattern on your canvas, you need either a larger canvas or a smaller pattern.

**Comment by gijdoul** [comment]
Hello,
no my pattern size is 3600x3600 and when I create a canvas and fill it with the pattern it's too big I mean the pattern is too big inside the new canvas I want to reduce the pattern size then when It fill the new canvas it look better

**Comment by gijdoul** [comment]
Here a screenshot to show you https://ibb.co/F7pDX79
the top image show the result of the command composite -tile cc.jpg -size 2550x1729  xc:none   mycanvas.jpg
and under it's on photoshop same size same pattern but pattern has been scaled down I want to get same result with imageMagick  like scale down my tile image to 30% or define a specific size whatever and then fill the new canvas hope it's more clear now :)

**Reply by snibgo** [reply]
So, you want to shrink your pattern before using it?
```
magick mypattern.jpg -resize 30% myp2.png
```
... then use myp2.png instead of mypattern.jpg.

**Reply by gijdoul** [reply]
I try it but get unable to open image 'myp2.png': No such file or directory @ error/blob.c/OpenBlob/3496.

I use this code: 
magick cc.jpg -resize 30% myp2.png composite -tile myp2.png -size 2550x1729 xc:none  mycanvas40.jpg

not sure I can scale down before tile the canvas on the fly.
because convert all my 3600x3600 to smaller size is not complicated but I want to do it in one go

**Comment by snibgo** [comment]
```
magick cc.jpg -resize 30% myp2.png composite -tile myp2.png -size 2550x1729 xc:none mycanvas40.jpg
```
That command is bad. I don't know what you want it to do, but it won't work. "composite" can't be used in the middle of a "magick" command.

**Comment by gijdoul** [comment]
maybe have a simple way to use my pattern at 30% of it size instead of resize first then use new pattern and make the canvas with but not sure it's possible

**Reply by fmw42** [reply]
try

`magick cc.jpg -resize 30% -write mpr:img +delete -size 2550x1729 tile:mpr:img result.png`

Does that do what you want?

**Comment by gijdoul** [comment]
I just try it work well only the resolution go down from 300dpi to 72 .
thanks a lot :)
Now I will try to fix the dpi quality, and for batch this command on a folder containing all my pattern it's another story 💯

**Reply by gijdoul** [reply]
I make it work for the batch using this command

for f in *.jpg; do convert "$f" -resize 30% -write mpr:img +delete -size 2550x1729 tile:mpr:img "${f%%.jpg}.png"; done
